shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -242,6 +243,41 @@ public static void ensureInitializationComplete(Context applicationContext, Stri
         }
     }
 
+    /**
+     * Same as {@link #ensureInitializationComplete(Context, String[])} but waiting on a background
+     * thread, then invoking {@code callback} on the {@code callbackHandler}.
+     */
+    public static void ensureInitializationCompleteAsync(
+        Context applicationContext,
+        String[] args,
+        Handler callbackHandler,
```

Comment:
When would the caller want to run `callback` on anything other than the main thread?

Reply:
One example of threading restrictions is that a SQL DB must be accessed on the same thread it was opened with. That specific example may not be highly relevant here, other than that sometimes things in Android come with strict threading requirements and so it is generally preferable to allow for that control. It's easy to hand over whatever Handler you want, it is tedious to get from another thread to the one you want.